server/core/src/administration_utilities.cpp

```diff
@@ -163,4 +167,9 @@ namespace irods
 
         return 0;
     } // remove_user
+
+    auto is_valid_zone_name(const std::string_view _zone_name) -> bool
+    {
+        return std::regex_match(std::begin(_zone_name), std::end(_zone_name), std::regex{R"(^[A-Za-z0-9\.]+$)"});
```

Comment:
Should std::regex be `static` so it's not built on every function call?

If this function is called often, I think it might be worth considering.

Reply:
I don't expect users to rename their zone often, if ever.

We recommend they spend time thinking of a good zone name during initial setup of the zone and sticking with it.

Reply:
However, given this is a free function, I'm now leaning toward making it static since we don't know how people will use it.

Will think about this a bit more.

Reply:
Changed it to be static.